frontend/src/components/placementVisuals/Placement.tsx

```diff
@@ -31,6 +37,57 @@ function Placement({ stepID, sampleIDs }: PlacementProps) {
     const destinationContainers = labworkStepPlacement.destinationContainers
     const activeSourceContainer = labworkStepPlacement.activeSourceContainer
     const activeDestinationContainer = labworkStepPlacement.activeDestinationContainer
+    const step = selectStepsByID(store.getState())[stepID]
+    const usesSamplesheet = step.name === EXPERIMENT_RUN_ILLUMINA_STEP
+    const cells = useAppSelector((state) =>  activeDestinationContainer?.name && selectContainer(state)({ name: activeDestinationContainer.name })?.cells)
+
+    const handleGetSamplesheet = useCallback(async () => {
+      type PlacementData = {
+          coordinates: string,
+          sample_id: FMSId,
+        }[]
+
+      if (!activeDestinationContainer) return
+      const placementData: PlacementData = []
+      try {
+        if (!cells) throw new Error(`Could not find active destination container in placement for '${activeDestinationContainer.name}'`)
+        for (const destinationCell of cells) {
+          if (!destinationCell.placedFrom) continue // cell does not contain sample placed from any source container
+          const sourceCell = selectCell(store.getState())(destinationCell.placedFrom)
+          if (!sourceCell) throw new Error(`Could not find cell at  ${destinationCell.placedFrom.coordinates}@${destinationCell.placedFrom.parentContainerName}`)
+          if (!sourceCell.sample) throw new Error(`There is no sample in source cell at ${destinationCell.placedFrom.coordinates}@${destinationCell.placedFrom.parentContainerName}`)
+          placementData.push({
+            coordinates: destinationCell.coordinates,
+            sample_id: sourceCell.sample,
+          })
+        }
+      } catch (e) {
+        notification.error({
+          message: e.message,
+          key: 'LabworkStep.Placement.GetSamplesheet',
+          duration: 20
+        })
+      }
+  
+      const fileData = await dispatch(api.samplesheets.getSamplesheet(activeDestinationContainer.barcode, activeDestinationContainer.kind, placementData))
+      if (fileData) {
+        downloadFromFile(fileData.filename, fileData.data)
+      }
+    }, [dispatch, activeDestinationContainer, cells])
+
+    const isPlacementComplete = useMemo(() => {
+      if (!activeDestinationContainer) return false
+      if (!cells) return false
+      const [Rows = [] as const, Columns = [] as const] = activeDestinationContainer.spec
+      const containerSize = Rows.length * Columns.length
+      let placedCells = 0
+      for (const destinationCell of cells) {
+        if (destinationCell.placedFrom) {
+          placedCells++
+        }
+      }
+      return placedCells === containerSize
+    }, [dispatch, activeDestinationContainer, cells])
```

Comment:
Unnecessary dispatch ;)
(according to linter)

Reply:
zzz